File: JSServer/DBCaller.js
function parseICS(file){
    var input = file.target;

    var reader = new FileReader();
    reader.onload = function(){
      var arrBuf = reader.result;
      //var output = document.getElementById('output');
      //output.src = dataURL;
      console.log(dataURL)
      console.log(arrayBuffer.byteLength);
    };
    reader.readAsArrayBuffer(input.files[0]);
    console.log(read)
  };

var openFile = function(event) {
    console.log("here")
    var input = event.target;
    var coveredSections = [];
    var reader = new FileReader();
    var textResults;
    reader.onload = function(){
        console.log("onload")
        textResults = reader.result;
        console.log("textResults")

        console.log(textResults);
        let arrayOfLines = textResults.match(/[^\r\n]+/g);
        console.log(arrayOfLines.length);
        var index = 0;
        var myBuffer = [];
        while(index < arrayOfLines.length){
            var line = arrayOfLines[index]
            if(line == "BEGIN:VEVENT") {
                console.log("recording event ... ")
                line = arrayOfLines[index]
                console.log(line)
                while(line != "END:VEVENT" && index < arrayOfLines.length) {
                    myBuffer.push(arrayOfLines[index]); 
                    index = index + 1  
                    line = arrayOfLines[index]
                }
                console.log(myBuffer.length);
                if(!(myBuffer.length == 8 || myBuffer.length == 9 ||  myBuffer.length == 10) ){
                    console.log('error in ics file')
                }
                coveredSections.push(AddSectionToDBFromCalendar(myBuffer));
                myBuffer = []
            }
            index = 1+index;
        }
        console.log(coveredSections)
        return coveredSections;
    };
    var text = reader.readAsText(input.files[0]);

};

function AddSectionToDBFromCalendar(section) {
    console.log("section",section)
    //There should be 7 lines of information that are used to decode the .ics file
    var data = []
    var summaryClass = section[1].replace("SUMMARY:", ""); //get rid of the "SUMMARY:" start

    var attendanceAndProfInfo = section[2]; //TODO: figure out how to get this information together
    var splitSections = attendanceAndProfInfo.split(";");
    var profName = splitSections[2].split(":")[1];

    var location = section[3].replace("LOCATION:", "");

    var startTime = section[4].replace("DTSTART:", "");
    var index = 5
    if(!section[5].includes(":")){
        startTime.concat(section[5])
        index = index+1
    }
    var endTime = section[index].replace("DTEND:", "");
    index = index+1
    if(!section[index].includes(":")){
        startTime.concat(section[5])
        index = index+1
    }
    startTime = startTime.substring(startTime.indexOf("T")+1);
    endTime = endTime.substring(startTime.indexOf("T")+1);

    var quarterAndYear = section[index].replace("CATEGORIES:","");

    var specificInfo = section[index];//should be basically a repeat of summaryClass, not sure we need this but I'll leave it here in case
    

    var prof = attendanceAndProfInfo.substring(attendanceAndProfInfo.indexOf(":") + 1);

    return {'SUMMARY' : summaryClass, 'PROFNAME': profName, "LOCATION": location, "STARTTIME": startTime, "ENDTIME": endTime, "QUARTERANDYEAR": quarterAndYear, "SPECIFICNFO": specificInfo, "PROF": prof}
};

File: ScheduleServer/src/DatabaseManagement/DBCaller.java
package DatabaseManagement;

import DataStructures.ClassManagement.ClassInfo;
import DataStructures.ClassManagement.Quarter;
import DataStructures.ClassManagement.Section;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.util.ArrayList;
import java.util.List;

public class DBCaller {

    public static void UploadCalendar(String filePath) throws Exception{
        ArrayList<Section> coveredSections = new ArrayList<>(); //sections we've already added to the calendar
                                                                //will be used when recording stuff

        //check for .ics file
        if (!filePath.endsWith(".ics")) {
            throw new Exception("invalid file, should be .ics file downloaded from schedule lookup");
        }
        File f = new File(filePath);
        BufferedReader reader = new BufferedReader(new FileReader(f));

        String st;
        while ((st = reader.readLine()) != null) {
            if(st.equals("BEGIN:VEVENT")) {
                System.out.println("Recording event...");

                List<String> buffer = new ArrayList<String>();
                String readingFile;
                while((readingFile = reader.readLine()) != "END:VEVENT") {
                    buffer.add(readingFile);
                }
                if(buffer.size() != 7) {
                    throw new Exception("Error with .ics file, buffer size not large enough");
                }
                coveredSections = AddSectionToDBFromCalendar(buffer, coveredSections);

            }
            System.out.println(st);

        }
    }

    private static ArrayList<Section> AddSectionToDBFromCalendar(List<String> buffer, ArrayList<Section> coveredSections) {
        //There should be 7 lines of information that are used to decode the .ics file
        String summaryClass = buffer.get(0).replace("SUMMARY:", ""); //get rid of the "SUMMARY:" start
        String attendanceAndProfInfo = buffer.get(1); //TODO: figure out how to get this information together
        String location = buffer.get(2).replace("LOCATION:", "");
        String startTime = buffer.get(3).replace("DTSTART:", "");
        String endTime = buffer.get(4).replace("DTEND:", "");
        String quarterAndYear = buffer.get(5).replace("CATEGORIES:","");
        String specificInfo = buffer.get(6);    //should be basically a repeat of summaryClass,
                                                // not sure we need this but I'll leave it here in case



        Section createdSection = new Section();
        ClassInfo q = new ClassInfo();


        return coveredSections;
    }


    private static boolean CheckIfQuarterExistsInDB(ClassInfo q){

        return false;
    }

}